src/modules/mavlink/mavlink_receiver.cpp

```diff
@@ -2273,14 +2274,16 @@ MavlinkReceiver::handle_message_heartbeat(mavlink_message_t *msg)
 }
 
 int
-MavlinkReceiver::set_message_interval(int msgId, float interval, int data_rate)
+MavlinkReceiver::set_message_interval(int msgId, float interval, float param3, float param4, int param5, int param6,
+				      int response_target)
 {
 	if (msgId == MAVLINK_MSG_ID_HEARTBEAT) {
 		return PX4_ERROR;
 	}
 
-	if (data_rate > 0) {
-		_mavlink.set_data_rate(data_rate);
+	if ((int)(param3 + 0.5f) || (int)(param4 + 0.5f) || param5 || param6 || response_target) {
+		// At least one of the unsupported params is non-zero
+		return PX4_ERROR;
```

Comment:
I'm just wondering if this could trigger when a param with NAN instead of 0 is sent and whether it's the right approach to respond with error in that case instead of just ignoring it.

Reply:
I want this to trigger if any of those params are non-zero, including if they were NaN.
But I think this is OK, because my understanding is that NaN evaluates as `false` in C++. 

> and whether it's the right approach to respond with error in that case instead of just ignoring it.

It would technically be an error because the default value for a reserved param in mavlink is defined as 0 if unspecified. If you want the default value to be NaN then you have to put that in the XML.

It could "really" matter in this case if we decided that NaN had meaning for those parameters other than "do nothing". That does happen sometimes. Making it a 0 check is safer.

Reply:
But I think this is OK, because my understanding is that NaN evaluates as false in C++.

Hm, I was assuming NAN + 0.5 is NAN and NAN cast to int will be some odd number which is not zero.

Let's see:
According to this it would be "undefined", which is not something we should rely on:
https://stackoverflow.com/questions/3986795/what-is-the-result-of-casting-float-inf-inf-and-nan-to-integer-in-c

In my own try, it seems to be -INT_MAX or 0, so very much undefined:
https://godbolt.org/z/Wd7qxM5T9